sound/soc/sof/ipc.c

```diff
@@ -270,9 +294,8 @@ int sof_ipc_tx_message(struct snd_sof_ipc *ipc, u32 header,
 	/* add message to transmit list */
 	list_add_tail(&msg->list, &ipc->tx_list);
 
-	/* schedule the message if not busy */
-	if (snd_sof_dsp_is_ready(sdev))
-		schedule_work(&ipc->tx_kwork);
+	/* try to send the message */
+	ipc_tx_next_msg_unlock(ipc);
```

Comment:
So, what happens if ipc_tx_next_msg_unlock fails to send the message, because DSP is already busy with another message?

Will tx_wait_done return an error right?